plugins/modules/rudder_server_settings.py

```diff
@@ -73,129 +72,161 @@
-                "Content-Type": "application/json",
-            }
-        else:
-            with open("/var/rudder/run/api-token") as f:
-                token = f.read()
-            self.headers = {
-                "X-API-Token": token,
-                "Content-Type": "application/json",
-            }
-
-        if module.params.get("rudder_url", None):
-            self.rudder_url = module.params.get("rudder_url")
-        else:
+        for param in allParams:
+            if param in module.params:
+                setattr(self, param, module.params[param])
+
+        if module.params.get('rudder_url', None) is None:
+            self.rudder_url = 'https://localhost/rudder'
             self.validate_certs = False
+        if module.params.get('rudder_token', None) is None:
+            with open('/var/rudder/run/api-token') as system_token:
+                token = system_token.read()
+        else:
+            token = module.params['rudder_token']
+        self.headers = {
+            'X-API-Token': token,
+            'Content-Type': 'application/json',
+        }
 
-        if module.params.get("validate_certs", None):
-            self.validate_certs = module.params.get("validate_certs")
+        raw_settings_to_set = {}
+        self.settings_to_set = self._translate_settings(raw_settings_to_set)
 
-    def _send_request(self, url, data=None, headers=None, method="GET"):
+    def _send_request(self, url, data=None, headers=None, method='GET'):
         if data is not None:
             data = json.dumps(data, sort_keys=True)
 
         if not headers:
             headers = []
 
-        full_url = "{rudder_url}{path}".format(
+        full_url = '{rudder_url}{path}'.format(
             rudder_url=self.rudder_url, path=url
         )
 
         resp, info = fetch_url(
             self._module, full_url, data=data, headers=headers, method=method
         )
 
-        status_code = info["status"]
+        status_code = info['status']
         if status_code == 404:
-            return None
+            self._module.fail_json(failed=True, msg='Not found')
         elif status_code == 401:
             self._module.fail_json(
                 failed=True,
                 msg="Unauthorized to perform action '{}' on '{}'".format(
                     method, full_url
                 ),
             )
-        elif status_code == 403:
-            self._module.fail_json(failed=True, msg="Permission Denied")
         elif status_code == 200:
             return self._module.from_json(resp.read())
         else:
             self._module.fail_json(
                 failed=True,
-                msg="Rudder API answered with HTTP {} details: {} ".format(
-                    status_code, info["msg"]
+                msg='Rudder API answered with HTTP {} details: {} '.format(
+                    status_code, info['msg']
                 ),
             )
 
+    def _translate_settings(self, settings_dict):
```

Comment:
This function does not really help for the settings. You just need to build a json with:
`{ "value": "setting_value" }`, the args can not be null since the parser is here to control that each call has a name and a value

Reply:
The allowed networks might be an exception, but we will see that later